### ISSUE TITLE
Script gets stuck at "WARNING Error while collecting events. Trying again..."

### ISSUE THREAD

**Issue by charnet3d** [philippnormann/chia-monitor#27]
Once in a while I found the script getting stuck at:

`WARNING Error while collecting events. Trying again...`

This message is generated from 2 places in ws_collector and rpc_collector. I haven't verified again which one it is in my case, I'll update once I have this info.

A workaround I did to have it running in a somewhat stable way, is to raise an `asyncio.CancelledError()` after the message `self.log.warning(f"Error while collecting events....")` so the process exits (hopefully.. I haven't dug deeper to make sure). And in addition to this chia-monitor is run from a .bat script that restarts it whenever it stops.

**Comment by charnet3d** [comment]
I did a run without my workaround and changed the error message to identify the origin and it seems it comes from ws_collector:

![image](https://user-images.githubusercontent.com/2863796/126281069-86b8ece9-20ef-49e1-9222-6e6bbb5e304a.png)

**Comment by philippnormann** [comment]
Hey @charnet3d, thanks for reporting this issue! I added more detailed error logging in https://github.com/philippnormann/chia-monitor/commit/bd67f3cd23bf7682f64a0625487035408994956d so we can uncover the root cause. Would you please retry using the latest version and post the error logs when this problem reoccurs?

**Comment by charnet3d** [comment]
Will do. Thanks.

**Comment by lopesmcc** [comment]
I've also been having WS issues for some days now - a minute or so after the monitor starts the WS connection closes (WS message type 257).

Not sure why it started happening, maybe it is related to the number of plots I have, and therefore some kind of rate-limit system on the daemon side?

```
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T10:56:26.707 WARNING Error while collecting events. Trying again... TypeError: Received message 257:None is not str
2021-07-28T1^C2021-07-28T10:56:26.716 INFO   🛑 Shutting down!
2021-07-28T10:56:26.717 ERROR  Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f56cb3f9ac0>
2021-07-28T10:56:26.717 ERROR  Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7f56ca2a00a0>, 556395.633289479)]']
connector: <aiohttp.connector.TCPConnector object at 0x7f56cb3f9ee0>
2021-07-28T10:56:26.717 INFO   👋 Bye!
2021-07-28T10:56:26.733 ERROR  Task exception was never retrieved
future: <Task finished name='Task-11' coro=<WsCollector.task() done, defined at /home/chia/chia-monitor/monitor/collectors/ws_collector.py:78> exception=KeyboardInterrupt()>
Traceback (most recent call last):
  File "/home/chia/chia-monitor/monitor/collectors/ws_collector.py", line 81, in task
    msg = await self.ws.receive_json()
  File "/home/chia/.local/share/virtualenvs/chia-monitor-sGgCuAIo/lib/python3.9/site-packages/aiohttp/client_ws.py", line 291, in receive_json
    data = await self.receive_str(timeout=timeout)
  File "/home/chia/.local/share/virtualenvs/chia-monitor-sGgCuAIo/lib/python3.9/site-packages/aiohttp/client_ws.py", line 276, in receive_str
    raise TypeError(f"Received message {msg.type}:{msg.data!r} is not str")
TypeError: Received message 257:None is not str
```

**Comment by lopesmcc** [comment]
I just saw this and think it is related:
```
TypeError: Received message 258:WebSocketError(<WSCloseCode.MESSAGE_TOO_BIG: 1009>, 'Message size 4635881 exceeds limit 4194304') is not str
```

**Comment by lopesmcc** [comment]
Fixed it by changing the `max_msg_size` on the WS client (to 50MB which is the max supported by the chia server).

**Comment by charnet3d** [comment]
Hi,

Thanks @lopesmcc for your input, I did find one error like the one you mentioned "TypeError: Received message 257:None is not str" once lately and probably your fix did it, not seeing it again for now.

However I encounter this other error (while the node is syncing, not sure if relevant) the RPC calls should be working. And also why does chia-monitor get stuck and not retry?

`
2021-07-29T11:29:06.335 [33mWARNING[0m Error while collecting events. Trying again... ConnectionError: Failed to get blockchain state via RPC. Is your full node running?[0m
`

**Comment by charnet3d** [comment]
I noticed that the msg "ConnectionError: Failed to get blockchain state via RPC. Is your full node running" was actually being generated because of a UNIQUE constraint being violated again.

In this specific case, I had 2 harvesters in a different network so they got recognized with the same gateway IP, so when trying to persist this event in the harvester events table using (ts, host) primary key it provokes the error.

In order to avoid these kinds of issues altogether I removed all the old primary keys and used autoincrement IDs like with the other tables you did @philippnormann.

I also included your change @lopesmcc in the Pull Request. Thank you.
